### ISSUE TITLE
help?

### ISSUE THREAD

**Issue by xiconet** [tusharmath/mtd-console#4]
Hi

"You can get started by using the --help option. This lists out all the possible operations in mtd."

```
mtd --help
To know more visit [https://github.com/tusharmath/mtd-console]
```

Is it supposed to be a joke?

**Comment by tusharmath** [comment]
@xiconet I agree it's not the best way, but I hope you were able to understand how to use?

**Comment by xiconet** [comment]
To answer your question: yes. But I think you really should make some "real" help available from inside the console, after all that's an important part of this kind of program philosophy. 
I was wondering if you left things as they are in a (quite legitimate) attempt to promote yourself. If true, I fear such method might be somewhat counter productive. 
Anyway, thanks for making your work available for free.

**Comment by tusharmath** [comment]
I feel sorry for you bro :D

The thing is, I am not able to get enough time to work on this project because I have been working on the core module here - https://github.com/tusharmath/multi-threaded-downloader/tree/develop

The console app is a thin wrapper around it and the `--help` param worked sometime back but I had to disable it because it din't make sense to have the documentation available here as well as in the main module. Everytime I had a new feature coming up, I had to update the console wrapper's documentation as well as the core lib's documentation, which was turning out to be a pain.

I agree this is not the best experience for the person who is using it but had this module gained enough popularity, I would worked on it more and tried to fix it.

**Comment by xiconet** [comment]
You really don't have to feel sorry for me, my dear Sir. After all your script _is_ documented. Besides I can always add some doc to the console script myself, in case I want to use it  without having to even open a browser in case I forgot how to use it. 
Of course I assume  you won't accept any pull request, since it appears you wouldn't consider it an enhancement, rather the contrary.
But again, my humble opinion is that it makes plenty of sense to have at least some usage reminders inside a console script, even if it may seem redundant.

Let's close this thread with the hope your work will gain popularity. 
Cheers.

**Comment by tusharmath** [comment]
The real problem is how I look at the future of this project. The console
wrapper would be a part of the main application.

If you send a pull request I would be more than happy to merge it. I am not
doing it for marketing purposes, I made it to solve a real problem, that
people like me who are in developing countries  - where internet speed is
low and expensive especially while downloading large files over http.

**Comment by xiconet** [comment]
All right then,  I'll be working on this in the next days

**Comment by tusharmath** [comment]
Thanks a ton!

**Comment by xiconet** [comment]
You're welcome. In France (where I'm currently living)  download speeds are (on average) not bad, nevertheless I find that it's often necessary to open multiple connections to get the most of them, except for some hosts who do not (artificially?) cap the max. speed per connection.
Just out of curiosity, what are the average internet speeds in your country?

**Comment by tusharmath** [comment]
Depends on where you are living in India, a small town could have something like 512kbps (thats when I built it) and now where I am living is a metropolitian city where the speed is 100Mbps at a much cheaper rate.
